### PR TITLE
Added Inferno HN

### PR DESCRIPTION
**Pull request by aretheregods**
InfernoJS implementation of Hacker News PWA.

Github Repo: https://github.com/aretheregods/hackerinferno
Running Application: https://infernohackernews.firebaseapp.com

I'm not sure what the weight values should be or if I listed the TTI correctly. The times listed on WPT and on the Lighthouse link from within WPT show different TTI. So, I listed the ones from the linked Lighthouse as the docs said. But I'm still not sure. Just so you know.

Please let me know if anything needs to be done or not.

### CODE REVIEW

**Comment by aretheregods** [comment]
I'm sure you guys are all super busy, as am I. But it would be awesome if I could get some feedback on how it's going with the pull request and what if anything I need to improve to join this little club of PWAs.

**Comment by usm4n** [comment]
I'll socond this. I am also waiting for the feed back on my PR. Thanks!

**Comment by dsolimando** [comment]
Idem

**Comment by addyosmani** [comment]
Thanks for the friendly ping on this PR. Apologies for the delay on reviewing. The community have been awesome filing all of these implementations. I'm going to go ahead and review this now.

File: site/_apps/inferno-hn.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+weight: 24
```
Comment (addyosmani) [inline site/_apps/inferno-hn.txt]: weight should now be `26`

File: site/_apps/inferno-hn.txt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+title: 'Inferno HN'
+github-title: 'aretheregods/hackerinferno'
+libraries:
+    - name: 'Inferno'
+scaffolding: 'Create-Inferno-App'
```
Comment (addyosmani) [inline site/_apps/inferno-hn.txt]: `create-inferno-app`

File: site/_apps/inferno-hn.txt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+service-worker: 'Application Shell + data caching with sw-precache'
+perfomance-patterns: 'PRPL, code-splitting for granular loading'
+server-side-rendering: 'None'
+api: 'HNPWA-API (unofficial)'
+hosting: 'Firebase functions over HTTP/2 with Server Push + edge-caching'
+other-details: 'Route-specific pushing'
+authors:
+    - name: 'Naythan'
+lighthouse: 91/100
+interactive-em: 2.6s
```
Comment (addyosmani) [inline site/_apps/inferno-hn.txt]: JavaScript is settling on the main thread in under 2.7 seconds here so this looks fine: https://www.webpagetest.org/video/compare.php?tests=171014_9P_6a3d5a195dc7b31e50e4fa94ef908639-r%3A1-c%3A0&thumbSize=200&ival=500&end=full

File: site/package-lock.json [hunk 4]
```diff
@@ -0,0 +1,1069 @@
+{
```
Comment (addyosmani) [inline site/package-lock.json]: You shouldn't need to update the package-lock file here. Please drop? :)

**Comment by addyosmani** [comment]
UX review:

- I checked top, new, show, jobs, ask. These views appear to render content correctly.
- You might see from the Lighthouse report that "Background and foreground colors do not have a sufficient contrast ratio." This may be due to the red on dark gray text that's currently in the UI. Can we address?

The stars and timestamps appear a little close together in entries:

image

I can into some occasional issues on the jobs page. I tried clearing the cache and refreshing and it was still an issue. It was unclear what caused this as jobs did initially show some content:

image

Views appear to work offline.

**Comment by chulian1819** [comment]
can you make it do server side rendering?

**Comment by aretheregods** [comment]
@addyosmani Thanks! I'm on these issues now.

**Comment by aretheregods** [comment]
@addyosmani I made the changes you mentioned. Hopefully everything is cool. If it isn't, please let me know. I also updated the times to reflect these changes. Although, the differences in loading times are probably more down to the time of day than anything with regard to the app, since I only made surface level updates.

**Comment by addyosmani** [comment]
@aretheregods The changes you've implemented look good. Thanks for working on them. The last task remaining to get this landed is improving the readability of text.

Right now, most of the main body text appears to use the same text size. This makes it trickier for the eye to distinguish between what is part of the title, what is the author etc.

image

Contrast to ReactHN or the YCombinator HN site:

image

image

My suggestion would be that we make the `detail-child` rows a smaller font and move the author (by foo) to the same row with this new smaller font-size. This will align it a little closer to the other apps.

I have no further feedback and we should be able to land it after that.

**Comment by aretheregods** [comment]
@addyosmani Thanks, I'm on it.

**Comment by aretheregods** [comment]
Sorry this took so long. I had to change the api I used on the backend, which affected the times a little bit. I updated the info to reflect that. Anyhow, let me know what's good.

**Comment by aretheregods** [comment]
@addyosmani I'm hoping I've fixed everything needed and can get this pulled. Just let me know if everything's okay.

**Comment by addyosmani** [comment]
Thanks once again for all your work here 👍

**Pull request merged**